Add documentation about file paths (#1857)
netlify/build: packages/config/docs/file_paths.md

> This document explains some of the main file paths used in a Netlify Site. This is mostly intended for code

Reviewer: [sand] I'd expect "site" to be lowercase here, since it's probably a general site and not a specific proper noun Site you're talking about. In fact, I see capitalized "Sites" in a few different places in this documentation. Unless it's a specific feature name, you can make those references lowercase.

Author: I tend to capitalize it to distinguish a Netlify Site from a website, which is a slightly different concept, since Netlify Sites include all Netlify features with them. However, if we prefer lowercasing it, then we should be consistent, so I've just fixed it. Thanks for the feedback! :+1:

contributors.

## Current directory

Process's current directory, as specified by `process.cwd()`.

This is mostly relevant for the Netlify CLI, where users can change it in their terminal.

## Repository root

Project root repository. In monorepos, this is the monorepo root.

This is the closest parent with a `.git` directory. If none is found, this defaults to the current directory.

## Build directory

This is used as the current directory when running the build command and Build plugins.

This is also used to resolve any file paths inside `netlify.toml` or the UI build settings, except for the `base`
configuration property (which is relative to the repository root instead).

We sometimes refer to this as the "base directory" or the "Site root" instead.

Its value is (from highest to lowest priority):

- The `base` property specified in `netlify.toml` or UI build settings. This is useful when targetting specific
  directories inside a monorepo.

> - In the Netlify CLI: if a project has multiple Netlify Sites, the user can step into it inside a terminal (using `cd`

Reviewer: > the user can step into it

I'm not sure exactly how to rephrase this since I'm a little unclear of the technical context, but I'd recommend using something other than a pronoun (it) to be a little clearer. 

Perhaps something like " In the Netlify CLI: If a project has multiple Netlify sites, the user can change directories to this build directory using the terminal."

Author: Makes sense, thanks!

  for example). The closest parent with a `netlify.toml` or `.netlify` will be used.
- The repository root.

## Configuration path

Path to the `netlify.toml`.

This is `undefined` if no `netlify.toml` exists.

This is relative to the current directory.

The following directories are searched (from highest to lowest priority):

- In the Netlify CLI: `--config` flag
- The build directory
- The current directory
- Any parent directory to the current directory

> If the `netlify.toml` specifies a `base` directory, the configuration file is searched a second time but using that
> `base` directory instead (\*).

Reviewer: I think may be missing a word, and I suggest a small edit at the end.
```suggestion
If the `netlify.toml` specifies a `base` directory, the configuration file is searched for a second time using the
`base` directory instead (\*).
```

## Absolute paths

The Netlify configuration never allows absolute paths. Any leading `/` in a path is omitted, resulting in a relative
path.

Prepending `./` is not required and does not do anything.

## baseRelDir

Boolean flag intended for backward compatibility. This is set for any Sites created since the end of 2019.

When unset:

- The points marked with (\*) above do not apply
- Paths in `netlify.toml` are relative to the repository root instead of the build directory

## Publish directory

> Directory where the Site built assets are output

Reviewer: Site's built assets? 
Site-buit assets?
(or lowercase site)
